mutation_writer/feed_writers.hh

```diff
@@ -63,15 +66,16 @@ future<> feed_writer(mutation_reader&& rd_ref, Writer wr) {
         }
     }
 
-    try {
-        co_await wr.close();
-    } catch (...) {
+    auto f = co_await coroutine::as_future(wr.close());
+    if (f.failed()) [[unlikely]] {
+        // Need to consume the failed future exception even if not used
+        auto close_ex = f.get_exception();
         if (!ex) {
-            ex = std::current_exception();
+            ex = std::move(close_ex);
         }
     }
-    if (ex) {
-        std::rethrow_exception(ex);
+    if (ex) [[unlikely]] {
+        co_await coroutine::return_exception_ptr(std::move(ex));
     }
 }
```

Comment:
In what way is this an optimization?

Reply:
IIUC, it saves the coroutine wrapper from catching the exception and converting it back to an exceptional future.
Unfortunately, the co_await'er will throw that exception, but we can't do much about it here.

Reply:
@avikivity are you OK with the patch?

Reply:
Yes, but is the error path important? We never hit it.

Reply:
but what's the advantage of keeping it as it is?
just less code churn, or is there anything else, like code footprint?

Reply:
Less code churn and less risk. The existing code looks simpler.

Reply:
kept `rethrow_exception` in v2 (373651c10b)